perun-core/src/main/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntry.java

```diff
@@ -660,6 +662,30 @@ public void setPerunBl(PerunBl perunBl) {
 		return getFacilitiesManagerBl().getHosts(sess, facility);
 	}
 
+	@Override
+	public List<EnrichedHost> getEnrichedHosts(PerunSession sess, Facility facility, List<String> attrNames) throws AttributeNotExistsException, FacilityNotExistsException {
+		Utils.checkPerunSession(sess);
+		getFacilitiesManagerBl().checkFacilityExists(sess, facility);
+
+		List<Host> hosts = getFacilitiesManagerBl().getHosts(sess, facility);
```

Comment:
You are not checking rights to get hosts for facility. You need to check it there or call method from api to correctly check rights.